deepsearch/documents/core/utils.py

```diff
@@ -0,0 +1,152 @@
+import datetime
+import requests
+from tqdm import tqdm
+import zipfile as z
+import glob
+import os
+from pathlib import Path
+import pathlib
+from typing import Any, List
+from .common_routines import progressbar_padding
+from deepsearch.cps.client.api import CpsApi
+
+
+class URLNavigator:
+    def __init__(self, api: CpsApi) -> None:
+        self.api = api
+        self.url_host = self.api.client.swagger_client.configuration.host
+        self.url_linked_ccs = (
```

Comment:
This looks like dangerous hackery, why are you reverse engineering the linked_ccs URL? It is well defined and one can easily use an URL parser to properly get the base host, path, etc.

I see that in other places you already used `urllib.parse.urlparse`